code/obj/machinery/turret.dm

```diff
@@ -126,9 +126,9 @@
 			continue
 		if (isdead(C) || isghostcritter(C))
 			continue
-		if (!istype(C.loc,/turf))
+		if (!(istype(C.loc,/turf) || istype(C.loc, /obj/vehicle)))
 			continue
-		if (!istype(C.loc.loc,A))
+		if (!(istype(C.loc.loc,A) || istype(C.loc.loc.loc, A)))
```

Comment:
this pains me to read

Reply:
is there any reason not to use `get_area()`?

Reply:
right, that's perfect; will add it!